pkg/authorization/authconfig.go

```diff
@@ -132,92 +133,103 @@ func (k *annotationAuthTypeDetector) Detect(_ context.Context, res *unstructured
 	return spi.Anonymous, nil
 }
 
-type expressionHostExtractor struct {
-	paths []string
-}
-
-func NewExpressionHostExtractor(paths []string) spi.HostExtractor {
-	return &expressionHostExtractor{paths: paths}
-}
+func UnifiedHostExtractor(extractors ...spi.HostExtractor) spi.HostExtractor { //nolint:gocognit //reason Inlined functions to avoid surface pollution
+	unique := func(in []string) []string {
```

Comment:
Small thing, but those names helped quickly mapped out what is going on:

```golang
unique := func(in []string) []string {
		set := map[string]bool{}

		for _, elem := range in {
			set[elem] = true
		}

		unique := make([]string, len(set))

		i := 0

		for elem := range set {
			unique[i] = elem
			i++
		}

		return unique
	}
```

Reply:
fixed https://github.com/maistra/odh-platform/pull/49/commits/7d94a7802a37cac651c1c3f32cef61e25e3840ce